backend/app/views/spree/admin/payments/_list.html.erb

```diff
@@ -57,12 +57,14 @@
             <% end %>
             <% allowed_actions = payment.actions.select { |a| can?(a.to_sym, payment) } %>
             <% allowed_actions.each do |action| %>
-              <% if action == 'credit' %>
+              <% if action == 'credit' && !payment.completed? %>
                 <%= link_to_with_icon 'reply', t('spree.refund'), new_admin_order_payment_refund_path(@order, payment), no_text: true %>
               <% elsif action == 'capture' && !@order.completed? %>
                 <%# no capture prior to completion. payments get captured when the order completes. %>
               <% else %>
-                <%= link_to_with_icon action, t(action, scope: 'spree'), fire_admin_order_payment_path(@order, payment, e: action), method: :put, no_text: true, data: {action: action} %>
+                <% if payment.void? || payment.invalid? %>
```

Comment:
I think this `if` condition is responsible for most (if not all) the failing specs, I think it should be different.

What's happening now is that, when all previous conditions are not met, then the action link is printed only if the payment is `void` or `invalid`.

I guess that among these actions that are filtered out by the conditional there's something that should be printed anyway. These actions come from this line:

```
<% allowed_actions = payment.actions.select { |a| can?(a.to_sym, payment) } %>
```

Reply:
@spaghetticode I just changed the approach, please let me know what do you think

Reply:
@jtapia sorry for the late feedback. I think it now may work properly. 

But I noticed there's no spec for this PR, and I think at some point in the past they were present, as some comments suggest. Not sure what happened to them, maybe they got lost during some rebase?